### ISSUE TITLE
Add support for Ubuntu 20.04 (Focal Fossa) release

### ISSUE THREAD

**Issue by AdnanHodzic** [Azure/azure-cli#13081]
**Is your feature request related to a problem? Please describe.**

`az cli` can't be installed on Ubuntu 20.04 using `apt` after referring to [official method](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-apt?view=azure-cli-latest), as right after `apt update` user will be greeted with following error:

```
E: The repository 'https://packages.microsoft.com/repos/azure-cli focal Release' does not have a Release file.
```

As I originally pointed out in https://github.com/Azure/azure-cli/issues/9585#issuecomment-608319542, only way to install it is to resort to a hack which is to add an older Ubuntu release to `/etc/apt/sources.list.d/azure-cli.list` file.  i.e: 

```
deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ disco main
```

Which will then lead to other problems as discovered in: https://github.com/Azure/azure-cli/issues/13030#issuecomment-615547689

**Describe the solution you'd like**

As someone who used to create various Debian packages and handle custom repositories. Since last Ubuntu release `libffi6` is [availability for is 19.04](https://packages.ubuntu.com/eoan/libffi6), I would simply repackage this package for 20.04 and upload it as such to `https://packages.microsoft.com/repos/azure-cli` repo. After adding support for `focal` release for the repository, you've solved the problem of inability to have `az cli` installed and all of its features fully working on >= Ubuntu 20.04.

**Describe alternatives you've considered**

I would encourage you as a company to give [az cli snap](https://snapcraft.io/azure-cli) more attention, as one such packaging format could've solve all the problem mentioned here. I had a similar problem with my Python app which I [addressed by creating a snap](https://github.com/AdnanHodzic/auto-cpufreq#snap-store).

**Additional context**

None, please inform me if you have any additional questions and/or comments.

**Comment by yonzhan** [comment]
@fengzhou-msft please take a look

**Comment by jetzerb** [comment]
My current workaround is a pair of hacks to install the latest pre-packaged version on focal / 20.04:
```sh
# 1. Override the installer script to hardcode `eoan` package source:
curl -sL https://aka.ms/InstallAzureCLIDeb | sed 's/${CLI_REPO}/eoan/g;' | bash;

# 2. Add a symlink to make it appear that libffi 6 is installed:
cd /lib/x86_64-linux-gnu;
ln -s libffi.so.7.1.0 libffi.so.6;
```

**Comment by hajekj** [comment]
> My current workaround is a pair of hacks to install the latest pre-packaged version on focal / 20.04:
> 
> ```shell
> # 1. Override the installer script to hardcode `eoan` package source:
> curl -sL https://aka.ms/InstallAzureCLIDeb | sed 's/${CLI_REPO}/eoan/g;' | bash;
> 
> # 2. Add a symlink to make it appear that libffi 6 is installed:
> cd /lib/x86_64-linux-gnu;
> ln -s libffi.so.7.1.0 libffi.so.6;
> ```

From WSL2 20.04 (upgraded from 18.04), I had to make the change in `/usr/lib/x86_64-linux-gnu`

**Comment by jetzerb** [comment]
@hajekj : Ah, yes.  I'm running az from inside a Linux Docker container, and Docker runs a Linux VM behind the scenes.  On a true Linux system, Ubuntu's /lib is a symbolic link to /usr/lib:
```
lrwxrwxrwx 1 root root 7 Apr 23 06:06 /lib -> usr/lib/
```
I don't think there's any translation of symlinks from *NIX filesystems to NTFS, so it's not surprising that you had to go to `/usr/lib` to find the file in WSL.

**Comment by frankfuu** [comment]
Also came across this issue. Is there a fix? I'm currently using `azure-cli` in a container as a temporary solution.

**Comment by 5m1** [comment]
Looking for any update on this as having same issue here

**Comment by jsturtevant** [comment]
Using the `eoan` install instructions above worked initially but then was unable to install az extensions properly.  I ended up falling back to the [script install](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-linux?view=azure-cli-latest#install-or-update).

**Comment by jklingen92** [comment]
I'm experiencing the same issue.

**Comment by ms-lolo** [comment]
Same issue here. The workarounds to get the package to install works but leaves you with an azure-cli that cannot install extensions. I've gotten things to work by creating my own python virutalenv and running `pip install azure-cli` instead since this doesn't try to do any weird distro version checks but this leaves me with only being able to run `az` commands when I am in that virtualenv so it's not an amazing workaround :(

**Comment by jetzerb** [comment]
@jsturtevant, @ms-lolo:  When I install `azure-cli` as described in my first comment above, I am able to install and use extensions.  When you say you can't install extensions, what do you mean?  It tells you that it can't?  It throws errors when you try to install extensions?  Extensions install but throw errors when you try to use them?
Wondering why I didn't have the same problem...Have you tried wading through the output of the install command and looking for errors?
```
az extension add --name azure-devops --debug
```
(feel free to install any other extension, `azure-devops` is just an example).

@jsturtevant: the method I posted above _is_ the script install you said you ended up falling back to.  Can you explain what you meant?  If you don't modify the script it'll error out because there is no focal package available.  And if you do modify it to pull a prior version, you still need to satisfy the `libffi` dependency.

**Comment by ms-lolo** [comment]
@jetzerb you're right. I was missing the `libffi.so.6` and adding that let extensions install fine. but I don't think I'm ok leaving that fix in place. I don't feel comfortable making that symlink part of the process of provisioning production machines. I'm going to hold off and keep using the virtualenv until the deb package is fixed.

**Comment by cubikca** [comment]
```
curl -sL https://aka.ms/InstallAzureCLIDeb | sudo ...
```
Not a habit you want to get into!

**Comment by aditzel** [comment]
I'm also running into this issue.

**Comment by owainharris** [comment]
Same issue here

**Comment by gauravthakur02** [comment]
I am also running into issues as my packages on the Ubuntu 20.04 LTS are failing to upgrade with the below error:
E: The repository 'https://packages.microsoft.com/repos/azure-cli focal Release' does not have a Release file.

Please do let us know the ETA when this will be available on Ubuntu 20.04 LTS.

**Comment by fengzhou-msft** [comment]
Focal package released for azure-cli 2.6.0. Please try again.

**Comment by mathieu-benoit** [comment]
FYI: I'm indeed able to install the 2.6.0 version on Ubuntu 20.4, via:
`curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash`. Thanks.

**Comment by AdnanHodzic** [comment]
> Focal package released for azure-cli 2.6.0. Please try again.

Thank you, I can confirm this adds proper support for Ubuntu 20.04.
